### ISSUE TITLE
Newly placed Signs not working correctly in POI.

### ISSUE THREAD

**Issue by FidgetyRat** [overviewer/Minecraft-Overviewer#1198]
My server noticed lately that new signs aren't being caught by our filtering rules. I saw in another thread to change from poi['Text1'] == "Location" to "Location" in poi['Text1']. Now the signs do show up, but they contain a lot of extra tagging and information!

>def locationFilter(poi):
    from fnr_additions import correctMinecraftText
    if poi['id'] == 'Sign' and "location" in correctMinecraftText(poi['Text1']).lower():
        return "\n".join([correctMinecraftText(poi['Text2']),
                          correctMinecraftText(poi['Text3']),
                          correctMinecraftText(poi['Text4'])])

Result:
>{extra:[Temporary],text:}
  {extra:[Ender Portal],text:}
  {extra:[Entrance],text:}

Note: The import just strips out some extra characters like quotations and such and works fine on our old signs.

Old signs parse just fine. I believe the "Old" signs were added during 1.8 with overviewer from 11/2014 and the new signs in 1.8.1 with overviewer from 01/08/2014. The only other changes are that we are also using Spigot on the server, but I can't comment as to whether spigot is adding the "Extra" stuff or not.

For now I added the "{extra:[" and "],text:}" to my library, so they get stripped out, but figured maybe this will hit someone else as well. For the longest time I couldn't figure out why my signs were not showing up at all.

**Comment by CounterPillow** [comment]
Seems like a duplicate of #1188

**Comment by Lomsor** [comment]
This isn't a duplicate of #1188 
I have the same issue. It stems from Spigot, with Vanilla this doesn't occur.
I looked into the NBT and the text is stored like this:
{"extra":["YOUR TEXT"],"text":""}

This only occurs on new signed placed while running Spigot 1.8

**Comment by CounterPillow** [comment]
You're right in that it's not an exact duplicate; spigot goes a step further to break things and puts the text in the "extra" field instead of the parent "text" field of the JSON, but in the end it still depends on #1188.

Properly solving #1188 will also solve this issue here though, since `"YOUR TEXT"` is considered a "valid" JSON object in this format.

So for all intents and purposes, it is a duplicate, as it properly conforms to the minecraft JSON text standard.

**Comment by j3rd** [comment]
A workaround using reg-ex, tested in python app not overviewer.

```python
import re

def signFilter(poi):
    if poi["id"] == "Sign":
        text = "\n".join([poi["Text1"], poi["Text2"], poi["Text3"], poi["Text4"]]).replace('"',"").decode("unicode-escape")
        return re.findall('(?<=\[)[^]]+(?=\])', text)
```

**Comment by CounterPillow** [comment]
regex is pretty much the most terrible way to do this.

**Comment by CounterPillow** [comment]
This (and consequentially #1188) have been resolved with a workaround which restores legacy behaviour of the sign text.

That being said, we can't do much about any issues which arise from the way Minecraft decided to do this. Certain 1.7 signs will be interpreted differently in Minecraft 1.8, and while Overviewer does its best to account for this, we can't do it correctly either due to a design decision Mojang has made.

**You can now again use the sign filter as demonstrated in the documentation, and do not need any other special regexing or replacements**. I do not plan on exposing any of the new sign information like colour or links to genPOI filters, as this goes beyond simply keeping genPOI alive until a better replacement is written.

Please report any problems you run into with signs, and I'll see what I can do. In the meantime, I'll also be thinking about ways to replace genPOI rather sooner than later, to make changes like this easier for all of us.
